code/game/machinery/constructable_frame.dm

```diff
@@ -59,27 +59,55 @@
 		return
 	switch(state)
 		if(1)
+			if(istype(P, /obj/item/weapon/circuitboard))
+				user << "<span class='warning'>The frame needs wiring first!</span>"
+				return
 			if(istype(P, /obj/item/stack/cable_coil))
 				var/obj/item/stack/cable_coil/C = P
 				if(C.get_amount() >= 5)
 					playsound(src.loc, 'sound/items/Deconstruct.ogg', 50, 1)
-					user << "<span class='notice'>You start to add cables to the frame.</span>"
-					if(do_after(user, 20))
+					user << "<span class='notice'>You start to add cables to the frame...</span>"
+					if(do_after(user, 20, target = src))
 						if(C.get_amount() >= 5 && state == 1)
 							C.use(5)
 							user << "<span class='notice'>You add cables to the frame.</span>"
 							state = 2
 							icon_state = "box_1"
 				else
-					user << "<span class='warning'>You need five length of cable to wire the frame.</span>"
+					user << "<span class='warning'>You need five length of cable to wire the frame!</span>"
 					return
+			if(istype(P, /obj/item/weapon/weldingtool) && !anchored)
+				var/obj/item/weapon/weldingtool/WT = P
+				if(WT.remove_fuel(0,user))
+					playsound(src.loc, 'sound/items/Welder2.ogg', 50, 1)
+					user.visible_message("<span class='warning'>[user] disassembles the frame.</span>", \
+										"<span class='notice'>You start to disassemble the frame...</span>", "You hear welding and clanking.")
+					if(do_after(user, 40, target = src))
+						if( !WT.isOn() )
+							return

```

Comment:
So just to make this clear, like this?
					if(do_after(user, 40, target = src))
						if( !WT.isOn() )
							return
						if(state == 1)
							user << "<span class='notice'>You disassemble the frame.</span>"
							var/obj/item/stack/sheet/metal/M = new (loc, 5)
							M.add_fingerprint(user)
							qdel(src)